libcodechecker/analyzers.py

```diff
@@ -0,0 +1,124 @@
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+"""
+Subcommand module for the 'CodeChecker analyzers' command which lists the
+analyzers available in CodeChecker.
+"""
+
+import argparse
+import subprocess
+
+from libcodechecker import generic_package_context
+from libcodechecker import output_formatters
+from libcodechecker.analyze.analyzers import analyzer_types
+from libcodechecker.logger import add_verbose_arguments
+from libcodechecker.logger import LoggerFactory
+
+LOG = LoggerFactory.get_new_logger('ANALYZERS')
+
+
+def get_argparser_ctor_args():
+    """
+    This method returns a dict containing the kwargs for constructing an
+    argparse.ArgumentParser (either directly or as a subparser).
+    """
+
+    return {
+        'prog': 'CodeChecker analyzers',
+        'formatter_class': argparse.ArgumentDefaultsHelpFormatter,
+
+        # Description is shown when the command's help is queried directly
+        'description': "Get the list of available and supported analyzers, "
+                       "querying their version and actual binary executed.",
+
+        # Help is shown when the "parent" CodeChecker command lists the
+        # individual subcommands.
+        'help': "List supported and available analyzers."
+    }
+
+
+def add_arguments_to_parser(parser):
+    """
+    Add the subcommand's arguments to the given argparse.ArgumentParser.
+    """
+
+    parser.add_argument('--all',
+                        dest="all",
+                        action='store_true',
+                        default=False,
+                        required=False,
+                        help="Show all supported analyzers, not just the "
+                             "available ones.")
+
+    parser.add_argument('--details',
+                        dest="details",
+                        action='store_true',
+                        default=False,
+                        required=False,
+                        help="Show details about the analyzers, not just "
+                             "their names.")
+
+    parser.add_argument('-o', '--output',
+                        dest='output_format',
+                        required=False,
+                        default='rows',
+                        choices=output_formatters.USER_FORMATS,
+                        help="Specify the format of the output list.")
+
+    add_verbose_arguments(parser)
+    parser.set_defaults(func=main)
+
+
+def main(args):
+    """
+    List the analyzers' basic information supported by CodeChecker.
+    """
+
+    context = generic_package_context.get_context()
+    working, errored = \
+        analyzer_types.check_supported_analyzers(
+            analyzer_types.supported_analyzers,
+            context)
+
+    if args.output_format not in ['csv', 'json']:
+        if not args.details:
```

Comment:
I'd offer a counterargument here: imagine reading the code. `--details` being specified is the _special_ and not the _general_ case &ndash; if you are searching in the code and skimming it with your eyes, having to find the _general_ case would, after a swap, require potentially a lot (here only a few lines, but otherwise potentially a lot) of LOC to skip over.

Same with the outside `if` (about `output_format`). `json` and `csv` is the more special cases here. I think it's more logical to have the _generic_ cases evaluated first.